ddpui/ddpairbyte/airbytehelpers.py

```diff
@@ -0,0 +1,50 @@
+from django.utils.text import slugify
+from ddpui.ddpairbyte import airbyte_service
+from ddpui.ddpairbyte.schema import AirbyteWorkspace
+from ddpui.ddpprefect import prefect_service
+from ddpui.ddpprefect import AIRBYTESERVER
+from ddpui.models.org import OrgPrefectBlock
+from ddpui.utils.ddp_logger import logger
+
+def setup_airbyte_workspace(wsname, org) -> AirbyteWorkspace:
+    """creates an airbyte workspace and attaches it to the org
+    also creates an airbyte server block in prefect if there isn't one already
+    we don't need to update any existing server block because it does not hold
+    the workspace id... only the connection details of the airbyte server
+    """
+
+    workspace = airbyte_service.create_workspace(wsname)
+
+    org.airbyte_workspace_id = workspace["workspaceId"]
```

Comment:
instead of calling workspace["workspaceId"] every time store it in the variable 

 workspace_id = workspace["workspaceId"]

Reply:
"every time"... you mean in the two places in this function??

Reply:
yes it's just better because think about this every time the code call it will call the function so instead of that we can store it in the variable and reference that value every time. Let me know if this is not clear.

Reply:
but it's not calling a function, it's doing a dictionary lookup

Reply:
workspace = airbyte_service.create_workspace(wsname) 

so this is returning the dictionary right so every time you're calling this workspace["workspaceId"] it's calling the function again? or am I wrong?

Reply:
the function is called once, the result of the function stored in a dictionary, the dict can be accessed as many times as you like the function will not be called again

You can verify this for yourself:

<img width="233" alt="image" src="https://user-images.githubusercontent.com/2160416/236603856-38a356ad-3c97-4ef6-8c3e-f3f6843eb134.png">